### PR TITLE
add check for etcd group

### PR DESCRIPTION
**Pull request by briandowns**
Signed-off-by: Brian Downs <brian.downs@gmail.com>

#### Proposed Changes ####

Adds check for etcd group when running in CIS mode.

#### Types of Changes ####

Adds a check for the etcd group and errors if it doesn't exist.

#### Verification ####

Run rke2 in cis mode with only the etcd user create. Notice the error indicating that the etcd group does not exist.

#### Linked Issues ####

#2288

### CODE REVIEW

File: pkg/cli/cmds/root.go [hunk 1]
```diff
@@ -245,7 +246,10 @@ func validateCISReqs(role CLIRole) error {
 	// etcd user only needs to exist on servers
 	if role == Server {
 		if _, err := user.Lookup("etcd"); err != nil {
-			ce = append(ce, fmt.Errorf("missing required %w", err))
+			ce = append(ce, errors.Wrap(err, "missing required"))
```
Comment (brandond) [inline pkg/cli/cmds/root.go:249]: I'll defer to @ShylajaDevadiga but I believe we also need to check that the `Gid` of the user returned by the `user.Lookup` call is the same as the `Gid` of the group from `user.LookupGroup`. If we ensure that here, the bits at https://github.com/rancher/rke2/blob/6f2632aaae172dda758813c1dc3c8928ba36682f/pkg/podexecutor/staticpod.go#L521 should fix the directory ownership for us.

**Review by brandond — CHANGES_REQUESTED** [review]

File: pkg/cli/cmds/root.go (hunk 1 above)
Comment (briandowns) [reply pkg/cli/cmds/root.go:249]: There's a chance that the uid and gid don't match depending on how the operator has created the groups. I just did a test where the uid and gid are different and user and groups are assigned correctly in the pod manifest as well as the applied ownership of the db directory.

Comment (brandond) [reply pkg/cli/cmds/root.go:249]: Ah, OK. The policy just says we need both an etcd user and group, and that the directory must be owned by them... but it doesn't have any requirement around the user's primary gid or membership in the etcd group. Fair enough.

**Review by brandond — APPROVED** [review]

File: pkg/cli/cmds/root.go (hunk 1 above)
Comment (ShylajaDevadiga) [reply pkg/cli/cmds/root.go:249]: uid and gid need not be the same while creating the user, we only need etcd user and etcd group to exist
etcd:x:990:987:etcd user:/home/etcd:/sbin/nologin

**Review by galal-hussein — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

**Pull request merged**